cmd/serverless-init/initcontainer/initcontainer.go

```diff
@@ -127,18 +127,12 @@ func handleSignals(
 			if sig != syscall.SIGCHLD {
 				if process != nil {
 					_ = syscall.Kill(process.Pid, sig.(syscall.Signal))
-					_, err := process.Wait()
```

Comment:
Do we still need to do this `process.Wait()`? Just curious why it was removed.

Reply:
i added a comment a bit late about this :)

Reply:
The reason why it was added in the first place was that the main process was terminating without waiting for the child process.
Removing the os.exit the main process will terminate once the child is done as expected https://github.com/DataDog/datadog-agent/blob/main/cmd/serverless-init/initcontainer/initcontainer.go#L91

Reply:
the only behavior that might be annoying is when:
- child process didn't start/terminate
- for some reason the main process is still alive
- sigterm is captured but it won't cause the main process to exit
since the lifecycle of the main process should be tied to the child process, I think it's better not to consider this scenario. If the main process lost the child and it's still running, that is a bug to be addressed